client/http-client/src/client.rs

```diff
@@ -134,6 +135,14 @@ impl HttpClientBuilder {
 		self
 	}
 
+	/// Set the HTTP(S) proxy that will proxy every HTTP request (default is none).
+	/// 
+	/// The proxy should be of the form <http[s]://host_or_ip:port> (without the brackets).
+	pub fn set_proxy(mut self, proxy: String) -> Self {
```

Comment:
would be better to return `Result<Self, Error>` and to parse proxy URI directly here.

```rust
	pub fn set_proxy(mut self, proxy: impl AsRef<str>) -> Result<Self, Error> {
      self.proxy = proxy.parse().map_err(|e| Error::Transport(....))?;
      self
   }
```